ecs/pod_creator_test.go

```diff
@@ -1,12 +1,249 @@
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName()).SetExecutionRole(testutil.ExecutionRole())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().
+				SetName(testutil.NewTaskDefinitionFamily("name")).
+				AddContainerDefinitions(*containerDef).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetTaskRole("role").
+				SetExecutionOptions(*execOpts)
+			assert.NoError(t, opts.Validate())
+
+			podCreator, err := NewBasicECSPodCreator(c, nil)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithIncompleteContainerDef": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image")
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().
+				SetName(testutil.NewTaskDefinitionFamily("name")).
+				AddContainerDefinitions(*containerDef).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetTaskRole("role").
+				SetExecutionOptions(*execOpts)
+			assert.NoError(t, opts.Validate())
+
+			podCreator, err := NewBasicECSPodCreator(c, v)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithSecretsButNoExecutionRole": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().SetName("envVar").
+				SetSecretOptions(*cocoa.NewSecretOptions().
+					SetName(testutil.NewSecretName("name")).
+					SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar}).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetName("container")
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().
+				SetCluster(testutil.ECSClusterName())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().
+				AddContainerDefinitions(*containerDef).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetTaskRole(testutil.TaskRole()).
+				SetExecutionOptions(*execOpts).
+				SetName(testutil.NewTaskDefinitionFamily("name"))
+			assert.NoError(t, opts.Validate())
+
+			podCreator, err := NewBasicECSPodCreator(c, v)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodSucceedsWithNewlyCreatedSecrets": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().
+				SetName("envVar").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName(testutil.NewSecretName("name")).SetValue("value"))
+			envVar.SecretOpts.SetExists(false)
```

Comment:
Nit: can you create the secret options like this?
```go
cocoa.NewSecretOptions().
    SetName(...).
    SetValue(...).
    SetExists(false).
    SetOwned(true)
```
The method spacing is optional, but it helps readability if you chain several method calls and put them on separate lines.

Reply:
Yep, but I called `SetOwned` on a separate line since it returns a `PodSecret` type, which isn't compatible with the `SecretOpts` type.